### PR TITLE
Convert Prepare, Extract to Systems

### PR DESCRIPTION
**Pull request by bigtimebuddy**
Converts the following Renderer and CanvasRenderer plugins to systems:

* Prepare
* Extract
* CanvasPrepare
* CanvasExtract

There are a few benefits for doing this:

* Get closer to disambiguating plugins as systems or pipelines
* These extensions are now strictly typed, plugin is just an any map

This change is backward-compatible with a deprecation warning.

### CODE REVIEW

File: packages/extract/test/Extract.tests.ts [hunk 1]
```diff
@@ -25,7 +26,7 @@ describe('Extract', () =>
     {
         const renderer = new Renderer();
         const sprite = new Sprite(Texture.WHITE);
-        const extract = renderer.plugins.extract as Extract;
+        const extract = renderer.extract as Extract;
```
Comment (Zyie) [inline packages/extract/test/Extract.tests.ts:29]: Testing locally I don't think these need to be cast anymore

```suggestion
        const extract = renderer.extract;
```

File: packages/extract/test/Extract.tests.ts [hunk 2]
```diff
@@ -39,7 +40,7 @@ describe('Extract', () =>
     it('should extract with no arguments', () =>
     {
         const renderer = new Renderer();
-        const extract = renderer.plugins.extract as Extract;
+        const extract = renderer.extract as Extract;
```
Comment (Zyie) [inline packages/extract/test/Extract.tests.ts:43]:
```suggestion
        const extract = renderer.extract;
```

File: packages/extract/test/Extract.tests.ts [hunk 3]
```diff
@@ -52,7 +53,7 @@ describe('Extract', () =>
     it('should extract a render texture', () =>
     {
         const renderer = new Renderer();
-        const extract = renderer.plugins.extract as Extract;
+        const extract = renderer.extract as Extract;
```
Comment (Zyie) [inline packages/extract/test/Extract.tests.ts:56]:
```suggestion
        const extract = renderer.extract;
```

**Comment by bigtimebuddy** [comment]
Follow-up to this will be making sure prepare and extract show up on the Renderer/CanvasRenderer documentation. Currently, docs are blocked because of an issue with webdocs that @ShukantPal is sorting.

**Pull request merged**